@xen-orchestra/fs/src/azure.js

```diff
@@ -0,0 +1,195 @@
+import { BlobServiceClient } from '@azure/storage-blob'
+import { createLogger } from '@xen-orchestra/log'
+import { parse } from 'xo-remote-parser'
+import { basename, join, split } from './path'
+import RemoteHandlerAbstract from './abstract'
+import { pRetry } from 'promise-toolbox'
+
+createLogger('xo:fs:azure')
+const MAX_BLOCK_SIZE = 1024 * 1024 * 1024 * 1024 * 4
+const MAX_BLOCK_COUNT = 1250
+const { warn } = createLogger('xo:fs:azure')
+
+// <http|https>://<account-name>.<service-name>.core.windows.net/<resource-path>
+
+export default class AzureHandler extends RemoteHandlerAbstract {
+  #container
+  #dir
+  #blobServiceClient
+  #containerClient
+
+  constructor(remote, _opts) {
+    super(remote)
+    const { username, path, password, host, protocol } = parse(remote.url)
+    /* 
+For Azure not Azurite
+ --------------------
+  const credentials = new StorageSharedKeyCredential(username, password);
+  this.#blobServiceClient = new BlobServiceClient(`${protocol}://${username}.blob.core.windows.net`, credentials);
+ --------------------
+*/
+
+    // this works ONLY for Azurite
+    this.#blobServiceClient = BlobServiceClient.fromConnectionString(
+      `DefaultEndpointsProtocol=${protocol};AccountName=${username};AccountKey=${password};BlobEndpoint=${protocol}://${host}/${username}`
+    )
+
+    const parts = split(path)
+    this.#container = parts.shift()
+    this.#dir = join(...parts)
+    this.#containerClient = this.#blobServiceClient.getContainerClient(this.#container)
+    this.#createContainer()
```

Comment:
you can't call an async method without a await or a then. And you can't call an async method in a constructor. Please move this  this in _sync()